### PR TITLE
Feature/searchable queue

### PR DESCRIPTION
**Pull request by joseferben**

### CODE REVIEW

File: sihl-queue/src/repo.ml [hunk 1]
```diff
@@ -30,6 +30,14 @@ module type Sig = sig
     -> unit
     -> Sihl.Contract.Queue.instance list Lwt.t
 
+  val search
```
Comment (joseferben) [inline sihl-queue/src/repo.ml:33]: @mabiede what do you think?

**Comment by joseferben** [comment]
I also adjusted the API to support serializing/deserializing of `ctx`. But I am not sure yet whether should do this or the user.

File: sihl/src/contract_queue.ml [hunk 2]
```diff
@@ -71,7 +74,7 @@ let should_run (job_instance : instance) now =
 let default_tries = 5
 let default_retry_delay = Core_time.Span.minutes 1
 
-let default_error_handler msg (instance : instance) =
+let default_error_handler ?ctx:_ msg (instance : instance) =
   Lwt.return
   @@ Logs.err (fun m ->
```
Comment (mabiede) [inline sihl/src/contract_queue.ml:79]: may add "pool" to log message if exists

```ocaml
let pool = 
  CCList.assoc_opt ~eq:(=) "pool"
  |> CCOption.map (fun pool -> Format.asprintf " (pool: '%s')" pool)
  |> CCOption.get_or ~default:""
```

or full context

```ocaml
(* with ppx_deriving.show *)
let context = Format.asprintf " (ctx: %s)" ([%show: (string * string) list] ctx) in
```

File: sihl-queue/src/repo.ml (hunk 1 above)
Comment (mabiede) [reply sihl-queue/src/repo.ml:33]: Looks fine, looking forward to test it in the project directly.

**Review by mabiede — CHANGES_REQUESTED** [review]

File: sihl-queue/test/queue.ml [hunk 3]
```diff
@@ -246,22 +281,26 @@ module Make (QueueService : Sihl.Contract.Queue.Sig) = struct
 
   let suite =
     [ ( "queue"
-      , [ test_case "should job run" `Quick should_run_job
+      , [ test_case "search jobs with tag" `Quick search
+        ; test_case "should job run" `Quick should_run_job
         ; test_case
-            "all dispatched jobs get processed"
+            "all\n             dispatched jobs get processed"
```
[the remark this answers is not in the document]
Comment (joseferben) [reply sihl-queue/test/queue.ml:287]: Fixed

File: sihl/src/contract_queue.ml (hunk 2 above)
Comment (joseferben) [reply sihl/src/contract_queue.ml:79]: Good point, thanks!

File: sihl/src/contract_queue.ml [hunk 4]
```diff
@@ -158,6 +162,16 @@ module type Sig = sig
     -> 'a job
     -> unit Lwt.t
 
+  (** TODO [josef] *)
```
[the remark this answers is not in the document]
Comment (joseferben) [reply sihl/src/contract_queue.ml:165]: Fixed :)

**Pull request merged**